analytics/src/analytics_android.cc

```diff
@@ -379,5 +381,60 @@ Future<std::string> GetAnalyticsInstanceIdLastResult() {
           internal::kAnalyticsFnGetAnalyticsInstanceId));
 }
 
+Future<int64_t> GetSessionId() {
+  FIREBASE_ASSERT_RETURN(Future<int64_t>(), internal::IsInitialized());
+  auto* api = internal::FutureData::Get()->api();
+  const auto future_handle =
+      api->SafeAlloc<int64_t>(internal::kAnalyticsFnGetSessionId);
+  JNIEnv* env = g_app->GetJNIEnv();
+  jobject task =
+      env->CallObjectMethod(g_analytics_class_instance,
+                            analytics::GetMethodId(analytics::kGetSessionId));
+
+  std::string error = util::GetAndClearExceptionMessage(env);
+  if (error.empty()) {
+    util::RegisterCallbackOnTask(
+        env, task,
+        [](JNIEnv* env, jobject result, util::FutureResult result_code,
+           const char* status_message, void* callback_data) {
+          int64_t session_id = 0;
+          if (result != nullptr) {
+            // result is a Long class type, unbox it.
+            session_id = util::JLongToInt64(env, result);
+          }
+          auto* future_data = internal::FutureData::Get();
+          if (future_data) {
+            bool success =
+                result_code == util::kFutureResultSuccess && result != nullptr;
+            FutureHandleId future_id =
+                reinterpret_cast<FutureHandleId>(callback_data);
+            FutureHandle handle(future_id);
+            future_data->api()->CompleteWithResult(
+                handle, success ? 0 : -1,
```

Comment:
Interesting. Analytics doesn't have an error code enum?

Reply:
No, it never had an error code until the instance ID getter which was a one-off until now. Adding an enum now would be an API change I think, but we should consider doing it if we ever get a third async method. :)

Reply:
Yeah, makes sense!